src/Sylius/Component/Promotion/Model/CatalogPromotion.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Promotion\Model;
+
+use Sylius\Component\Resource\Model\ResourceInterface;
+
+class CatalogPromotion implements ResourceInterface
+{
+    /** @psalm-suppress PropertyNotSetInConstructor */
+    protected int $id;
+    /** @psalm-suppress PropertyNotSetInConstructor */
+    protected string $name;
+    /** @psalm-suppress PropertyNotSetInConstructor */
```

Comment:
I totally understand we need it, but in my perfect world I would like to keep PHP classes as configuration-free as possible 😄 Maybe putting it in the psalm.xml file would work? (on the other hand, it could result in a giga-file, impossible to manage 🤷‍♂️ )

Reply:
Yup, I agree we need to think about how to make it better.

Reply:
Isn't nulling a way to go? Also, as this is resource model, these fields will be nullable anyway